modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/MetricNameValidator.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.telemetry.apm.internal;
+
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class MetricNameValidator {
+    private final Pattern ALLOWED_CHARACTERS = Pattern.compile("[a-z][a-z0-9_]*");
+    private final Set<String> LAST_ELEMENT_ALLOW_LIST = Set.of(
+        "size",
+        "total",
+        "count",
+        "usage",
+        "utilization",
+        "histogram",
+        "ratio",
+        "status" /*a workaround for enums */,
+        "time"
+    );
+
+    static final int MAX_ELEMENT_LENGTH = 30;
+
+    public void validate(String name) {
+        String[] elements = name.split("\\.");
```

Comment:
Is there overall metric name length limit? (not one per element)

Reply:
yes, it is 255characters. It is being verified by an agent itself

Reply:
I think it would be better to have it here too, so we can fail here consistently